stopes/ui/seamlisten/backend/app/fileviewer.py

```diff
@@ -131,7 +131,14 @@ def get_annotations(
 @router.post("/general/")
 async def general_query(query: DefaultQuery) -> Response:
     query_path = query.gz_path.strip()
-    print(query_path.split(" "))
+    # Convert the path to an absolute path
+    query_path = Path(os.path.expanduser(query_path)).resolve()
+    # Check if the path is a directory
+    print(query_path)
+    if query_path.is_dir():
+        result_data = gather_folder_contents(query_path)
+        return result_data
+
     if query_path.endswith(".tsv.gz") or query_path.endswith(".zip"):
```

Comment:
I don't think that Path has an `endswith` API (https://devdocs.io/python~3.8/library/pathlib#pathlib.Path), maybe use `suffixes` or cast it to string first.

Reply:
same on line 148, `split` will not work. Path might not even work properly there.

Maybe make keep a Path version and the original str version to do this.